src/main/java/org/osc/controller/nsc/api/SampleSdnRedirectionApi.java

```diff
@@ -0,0 +1,329 @@
+                    inspectionPortId), e); // TODO
+            return null;
+        }
+    }
+
+    @Override
+    public String installInspectionHook(List<NetworkElement> inspectedPorts, InspectionPortElement inspectionPort,
+            Long tag, TagEncapsulationType encType, Long order, FailurePolicyType failurePolicyType)
+            throws NetworkPortNotFoundException, Exception {
+
+        if (inspectedPorts != null && inspectedPorts.size() > 0) {
+
+            // TODO : must loop through the engire list
+            NetworkElement inspected = inspectedPorts.get(0);
+            String retVal = null;
+
+            retVal = this.txControl.required(() -> {
+
+                NetworkElement ingress = inspectionPort.getIngressPort();
+                NetworkElement egress = inspectionPort.getEgressPort();
+                InspectionPortEntity inspectionPortTmp = this.utils.findInspPortByNetworkElements(ingress, egress);
+                InspectionHookEntity inspectionHookEntity = this.utils.findInspHookByInspectedAndPort(inspected, inspectionPortTmp);
+
+                if (inspectionHookEntity == null) {
+                    // TODO : should we also merge egress and ingress?
+                    if (inspectionPortTmp == null) {
+                        LOG.warn("InspectionPort not found creating InspectionHook!");
+                    }
+
+                    inspectionHookEntity = this.utils.makeInspectionHookEntity(inspected, inspectionPortTmp, tag,
+                                                    encType, order, failurePolicyType);
+                }
+
+                inspectionHookEntity = this.em.merge(inspectionHookEntity);
+                String hookId = inspectionHookEntity.getHookId();
+                return hookId;
+            });
+
+            return retVal;
+        } else {
+            LOG.error("Attempt to install inspection hook with no inspected or inspection port!");
+            return null;
+        }
+    }
+
+    @Override
+    public void removeInspectionHook(List<NetworkElement> inspectedPort, InspectionPortElement inspectionPort)
+            throws Exception {
+
+        if (inspectedPort != null && inspectedPort.size() > 0) {
+
+            NetworkElement inspected = inspectedPort.get(0);
+
+            this.txControl.requiresNew(() -> {
+                InspectionHookEntity entity = this.utils.findInspHookByInspectedAndPort(inspected, inspectionPort);
+                if (entity != null) {
+                    this.utils.removeSingleInspectionHook(entity);
+                }
+                return null;
+            });
+        }
+    }
+
+    @Override
+    public Long getInspectionHookTag(NetworkElement inspectedPort, InspectionPortElement inspectionPort)
+            throws NetworkPortNotFoundException, Exception {
+        InspectionHookElement inspectionHook = getInspectionHook(inspectedPort, inspectionPort);
```

Comment:
Can this method also use findInspHookByInspectedAndPort for consistency with the one below? You dont need the *Element to return the tag, you can return from the entity.